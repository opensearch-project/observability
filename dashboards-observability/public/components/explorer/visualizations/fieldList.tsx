/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { FieldsAccordion } from './fields_accordion';

export const FieldList = (
  {
    fields,
    id
  }: any
) => {
  
  return (
    <div
      className="lnsIndexPatternFieldList"
    >
      <div className="lnsIndexPatternFieldList__accordionContainer">
        {
          <>
            <FieldsAccordion
              id={id}
              paginatedFields={ fields || [] }
              isFiltered={ false }
              label={ "Available fields" }
              showExistenceFetchError={ false }
            />
          </>
        }
      </div>
    </div>
  );
};