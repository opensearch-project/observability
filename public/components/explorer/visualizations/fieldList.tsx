/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

// import './field_list.scss';
import { throttle } from 'lodash';
import React, { useState, Fragment, useCallback, useMemo, useEffect } from 'react';
import { EuiSpacer } from '@elastic/eui';
import { FieldItem } from './field_item';
import { FieldsAccordion } from './fields_accordion';

export const FieldList = (
  {
    schema,
    id
  }
) => {

  return (
    <div
      className="lnsIndexPatternFieldList"
    >
      <div className="lnsIndexPatternFieldList__accordionContainer">
        {
          <>
            <FieldsAccordion
              id={id}
              paginatedFields={ schema }
              isFiltered={ false }
              label={ "Available fields" }
              showExistenceFetchError={ false }
            />
          </>
        }
      </div>
    </div>
  );
};