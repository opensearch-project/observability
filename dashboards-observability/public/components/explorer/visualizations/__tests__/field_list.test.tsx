/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { waitFor } from '@testing-library/react';
import { FieldList } from '../fieldList';
import { 
  AVAILABLE_FIELDS
} from '../../../../../test/event_analytics_constants';

describe('Visualization field list component', () => {
  configure({ adapter: new Adapter() });

  it('Renders field list component', async () => {
    
    const wrapper = mount(
      <FieldList
        id="3327"
        field={AVAILABLE_FIELDS}
      />
    );
    
    wrapper.update();

    await waitFor(() => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
