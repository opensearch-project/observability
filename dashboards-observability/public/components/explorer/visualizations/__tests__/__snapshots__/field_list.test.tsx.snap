// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Visualization field list component Renders field list component 1`] = `
<FieldList
  field={
    Array [
      Object {
        "name": "agent",
        "type": "string",
      },
      Object {
        "name": "bytes",
        "type": "long",
      },
      Object {
        "name": "clientip",
        "type": "ip",
      },
      Object {
        "name": "event",
        "type": "struct",
      },
      Object {
        "name": "extension",
        "type": "string",
      },
      Object {
        "name": "geo",
        "type": "struct",
      },
      Object {
        "name": "host",
        "type": "string",
      },
      Object {
        "name": "index",
        "type": "string",
      },
      Object {
        "name": "ip",
        "type": "ip",
      },
      Object {
        "name": "machine",
        "type": "struct",
      },
      Object {
        "name": "memory",
        "type": "double",
      },
      Object {
        "name": "message",
        "type": "string",
      },
      Object {
        "name": "phpmemory",
        "type": "long",
      },
      Object {
        "name": "referer",
        "type": "string",
      },
      Object {
        "name": "request",
        "type": "string",
      },
      Object {
        "name": "response",
        "type": "string",
      },
      Object {
        "name": "tags",
        "type": "string",
      },
      Object {
        "name": "timestamp",
        "type": "timestamp",
      },
      Object {
        "name": "url",
        "type": "string",
      },
      Object {
        "name": "utc_time",
        "type": "timestamp",
      },
    ]
  }
  id="3327"
>
  <div
    className="lnsIndexPatternFieldList"
  >
    <div
      className="lnsIndexPatternFieldList__accordionContainer"
    >
      <Memo(InnerFieldsAccordion)
        id="3327"
        isFiltered={false}
        label="Available fields"
        paginatedFields={Array []}
        showExistenceFetchError={false}
      >
        <EuiAccordion
          arrowDisplay="left"
          buttonContent={
            <EuiText
              size="xs"
            >
              <strong>
                Available fields
              </strong>
            </EuiText>
          }
          extraAction={
            <EuiNotificationBadge
              color="subdued"
              size="m"
            >
              0
            </EuiNotificationBadge>
          }
          id="3327"
          initialIsOpen={false}
          isLoading={false}
          isLoadingMessage={false}
          paddingSize="none"
        >
          <div
            className="euiAccordion"
          >
            <div
              className="euiAccordion__triggerWrapper"
            >
              <button
                aria-controls="3327"
                aria-expanded={false}
                className="euiAccordion__button"
                id="random_html_id"
                onClick={[Function]}
                type="button"
              >
                <span
                  className="euiAccordion__iconWrapper"
                >
                  <EuiIcon
                    className="euiAccordion__icon"
                    size="m"
                    type="arrowRight"
                  >
                    <EuiIconEmpty
                      aria-hidden={true}
                      className="euiIcon euiIcon--medium euiIcon-isLoading euiAccordion__icon"
                      focusable="false"
                      role="img"
                      style={null}
                    >
                      <svg
                        aria-hidden={true}
                        className="euiIcon euiIcon--medium euiIcon-isLoading euiAccordion__icon"
                        focusable="false"
                        height={16}
                        role="img"
                        style={null}
                        viewBox="0 0 16 16"
                        width={16}
                        xmlns="http://www.w3.org/2000/svg"
                      />
                    </EuiIconEmpty>
                  </EuiIcon>
                </span>
                <span
                  className="euiIEFlexWrapFix"
                >
                  <EuiText
                    size="xs"
                  >
                    <div
                      className="euiText euiText--extraSmall"
                    >
                      <strong>
                        Available fields
                      </strong>
                    </div>
                  </EuiText>
                </span>
              </button>
              <div
                className="euiAccordion__optionalAction"
              >
                <EuiNotificationBadge
                  color="subdued"
                  size="m"
                >
                  <span
                    className="euiNotificationBadge euiNotificationBadge--medium euiNotificationBadge--subdued"
                  >
                    0
                  </span>
                </EuiNotificationBadge>
              </div>
            </div>
            <div
              className="euiAccordion__childWrapper"
              id="3327"
            >
              <EuiResizeObserver
                onResize={[Function]}
              >
                <div>
                  <div
                    className=""
                  >
                    <EuiSpacer
                      size="s"
                    >
                      <div
                        className="euiSpacer euiSpacer--s"
                      />
                    </EuiSpacer>
                    <div
                      className="lnsInnerIndexPatternDataPanel__fieldItems"
                    />
                  </div>
                </div>
              </EuiResizeObserver>
            </div>
          </div>
        </EuiAccordion>
      </Memo(InnerFieldsAccordion)>
    </div>
  </div>
</FieldList>
`;
